optuna/visualization/_utils.py

```diff
@@ -99,3 +103,26 @@ def _get_param_values(trials: List[FrozenTrial], p_name: str) -> List[Any]:
     if _is_numerical(trials, p_name):
         return values
     return list(map(str, values))
+
+
+def _filter_inf_trials(trials: List[FrozenTrial]) -> List[FrozenTrial]:
+
+    filtered_trials: List[FrozenTrial] = []
+    for trial in trials:
+        intermediate_values = list(trial.intermediate_values.values())
+        if trial.value is not None and np.isinf(trial.value):
+            _logger.info(
+                f"Trial {trial.number} is omitted in visualization ",
+                "because its objective value is inf.",
+            )
+
+        elif any(np.isinf(intermediate_values)):
+            _logger.info(
+                f"Trial {trial.number} is omitted in visualization ",
+                "because one or more intermediate values are inf.",
+            )
```

Comment:
In my understanding, `intermediate_values` is used only by `plot_intermediate_values` methods in `optuna.visualization.*.py` and current plot methods can deal inf/nan correctly. So we might not need this skip part for intermediate values. I confirm that by running the following code:

```python
import numpy as np
import optuna

# nan case
def objective(trial):

    for step in range(10):
        if step == trial.number:
            trial.report(np.nan, step)
        else:
            trial.report(trial.number, step)
    return 1.

study = optuna.create_study()
study.optimize(objective, n_trials=10)
optuna.visualization.plot_intermediate_values(study)
optuna.visualization.matplotlib.plot_intermediate_values(study)

# inf case
def objective(trial):

    for step in range(10):
        if step == trial.number:
            trial.report(np.inf, step)
        else:
            trial.report(trial.number, step)
    return 1.

study = optuna.create_study()
study.optimize(objective, n_trials=10)
study = optuna.create_study()
study.optimize(objective, n_trials=10)
optuna.visualization.plot_intermediate_values(study)
optuna.visualization.matplotlib.plot_intermediate_values(study)
```

Definitely, my poor issue description asked you to implement this. Sorry for that...

Reply:
No worries, I missed how plotly/matplotlib backends handle this case as well. Double checked using the test case you provided and indeed both backends render plots with missing datapoints where inf/nan were, which i guess is as valid as removing such trials altogether. Removed unnecessary code path and test in fbef410d801df8281f072d8c200b624161df2c50.